app/controllers/devise/saml_sessions_controller.rb

```diff
@@ -16,6 +16,7 @@ def new
     request = OneLogin::RubySaml::Authrequest.new
     auth_params = { RelayState: relay_state } if relay_state
     action = request.create(saml_config(idp_entity_id), auth_params || {})
+    session[:saml_transaction_id] = request.request_id
```

Comment:
This persists the SAML request ID in the user's session regardless of whether or not the service provider has opted-in via the configuration. That seems pretty harmless to me, and I'm hoping it will avoid unexpected validation errors right after service providers opt-in.